### DISCUSSION TITLE
.Net Maui Support

### DISCUSSION THREAD

**Discussion by itsKXCode** [emgucv/emgucv#727] [Q&A]
Will .Net Maui be supported in the Future?

**Comment by emgucv** [comment]
We will look into .Net Maui once it is officially released. 

In theory you should be able to use our nuget packages in .Net Maui without any issues.

**Reply by ambirdev** [reply]
.NET Maui can access .NET Standard 2.0 and thus if you build to .NET Standard 2.0, is accessible from .NET Maui on Windows.  Not sure about Mac and other platforms.

**Comment by vmgolubev** [comment]
Hi, @emgucv, any progress on that? I bought EmguCV for Droid and iOS, Droid NuGet packages work like a charm in MAUI app, but the iOS package does not, the error is:
Restoring packages ....App.csproj...
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-android31.0 (.NETCoreApp,Version=v6.0). Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-ios15.4 (.NETCoreApp,Version=v6.0). Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-android31.0 (.NETCoreApp,Version=v6.0) / android-arm. Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-android31.0 (.NETCoreApp,Version=v6.0) / android-arm64. Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-android31.0 (.NETCoreApp,Version=v6.0) / android-x64. Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-android31.0 (.NETCoreApp,Version=v6.0) / android-x86. Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-android31.0 (.NETCoreApp,Version=v6.0) / ios-arm64. Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-ios15.4 (.NETCoreApp,Version=v6.0) / android-arm. Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-ios15.4 (.NETCoreApp,Version=v6.0) / android-arm64. Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-ios15.4 (.NETCoreApp,Version=v6.0) / android-x64. Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-ios15.4 (.NETCoreApp,Version=v6.0) / android-x86. Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
NU1202: Package Emgu.CV.runtime.ios 4.5.5.4823 is not compatible with net6.0-ios15.4 (.NETCoreApp,Version=v6.0) / ios-arm64. Package Emgu.CV.runtime.ios 4.5.5.4823 supports: xamarinios10 (Xamarin.iOS,Version=v1.0)
Package restore failed. Rolling back package changes for 'App'.
Time Elapsed: 00:00:00.9677369
========== Finished ==========

This project has the Droid package and EmguCV package installed, but the installation of the EmguCV.iOS package is not possible. How can that be fixed?
Best,
V

**Reply by softlion** [reply]
With Maui, the iOS runtime has breaking changes compared to Xamarin. So EMGU has to be build specifically against the new runtime (runtime identifier: net6-ios).

On Android it should be rebuilt for net6-android instead of xamarin.android too. As there is no breaking change it may work with the old runtime identifier (xamarin.android), but it's not guaranteed.

**Comment by emgucv** [comment]
.net MAUI is actually a completely different platform compares with Xamarin. You will need a different nuget package to target .net MAUI. 

We are preparing the new 4.6.0 release with .NET MAUI targeted nuget packages for Android, iOS and Mac OS. For windows, you can use the same runtime package in MAUI. If you want to get early access to either of the above, please email "support@emgu.com" and we can reply with a development built.

**Comment by emgucv** [comment]
4.6.0 release is out with experimental support for MAUI. Commercial license user can download the maui nuget package by following this instruction:
https://www.emgu.com/wiki/index.php/Download_And_Installation#Commercial_Release
